New options

MeuhV1 opened the pull request:
- Added some new options: No Human Stability, No stability all civs, unlimited switching (All off by default)
- All options can be changed during the game in the WB (gameoptions menu)
- Added function to enable/disable secondary civs in the gameoption menu
- Moved setPlayerEnabled function to StoredData.py (It was in several places. Now it is in just one place)

Several people asked how to to modify their game to incorporate these changes. These new options make it a lot easier for them.

Commit:
- Added some new options: No Human Stability, No stability all civs, unlimited switching (All off by default)
- All options can be changed during the game in the WB (gameoptions menu)
- Added function to enable/disable secondary civs in the gameoption menu
- Moved setPlayerEnabled function to StoredData.py (It was in several places. Now it is in just one place)

dguenms commented:
Good idea.

The only point I would make is to rename the two no stability options to no collapse, because the stability mechanic works regardless of the setting as far as I can tell. Alternatively, you can also abort all stability calculations with the corresponding options set - this should also save some computational power.

MeuhV1 commented:
I'm aware that the calcuations still run in the background. I did consider aborting calculating all stability calcuations, but I did the easy, lazy option. (And I didn't want to miss a trigger, because I was afraid the would cause errors)

But now I see the easy way to disable the calculations. Just add a return statement to all functions triggered by other files which do stability calculations in stability.py. That should do it, right?

dguenms commented:
The way I see it the central component of Stability.py (and also the most computationally expensive) is calculateStability(). Just adding a "return 0" right at the beginning if the option is selected should save a lot of time and not break anything. Leave the current returns for the collapse checks in place to be sure, but otherwise I don't think you have to mess with the triggers, their contribution to the calculation time is probably minimal.

Commit:
Moved "already switched" option to a new list ("Stored variables")
Updated some wording of the DoC options screen
"No Stability" option completely prevents stability calculations and parameter updating

MeuhV1 commented:
Updated the PR.

I thought of adding a new option which allows to change the stability level of each civ in the WB. I decided not to add it, because your stability level will return to the old situation after a couple of checks. So it doesn't have a significant impact.

dguenms commented on Assets/Python/StoredData.py:
```diff
@@ -150,6 +153,24 @@ def getTimedConquests(self):
+	def setPlayerEnabled(self, iCiv, bNewValue):
+		self.scriptDict['lPlayerEnabled'][con.lSecondaryCivs.index(iCiv)] = bNewValue
+		
+	def getPlayerEnabled(self, iCiv):
```
Just a small comment, by convention getter methods returning a boolean are usually called isPlayerEnabled() etc., same for the other ones here.

dguenms commented:
Looks good!

Commit:
Changed getXXX() functions to isXXX() functions

MeuhV1 commented:
Updated

dguenms commented:
Great, merged.

Commit:
2 new options: No Congress, No Plague
Moved StoredData functions from Plague.py to StoredData.py
New editable variable: Amount of turns until next congress
